internal/manifest/oci_index.go

```diff
@@ -84,6 +84,45 @@ func (index *OCI1IndexPublic) UpdateInstances(updates []ListUpdate) error {
 	return nil
 }
 
+func (index *OCI1Index) EditInstances(updates []ListEdit) error {
+	if len(updates) != len(index.Manifests) {
+		//return fmt.Errorf("incorrect number of update entries passed to OCI1Index.UpdateInstances: expected %d, got %d", len(index.Manifests), len(updates))
+		if len(updates) > len(index.Manifests) {
+			newIndexes := make([]imgspecv1.Descriptor, len(updates))
+			copy(newIndexes, index.Manifests)
+			index.Manifests = newIndexes
+		}
+	}
+	for i, editInstance := range updates {
+		switch editInstance.op {
+		case op_update:
+			if err := updates[i].Digest.Validate(); err != nil {
+				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances contained an invalid digest: %w", i+1, len(updates), err)
+			}
+			index.Manifests[i].Digest = updates[i].Digest
+			if updates[i].Size < 0 {
+				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had an invalid size (%d)", i+1, len(updates), updates[i].Size)
+			}
+			index.Manifests[i].Size = updates[i].Size
+			if updates[i].MediaType == "" {
+				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had no media type (was %q)", i+1, len(updates), index.Manifests[i].MediaType)
+			}
+			index.Manifests[i].MediaType = updates[i].MediaType
+			if updates[i].Platform != nil {
+				index.Manifests[i].Platform = updates[i].Platform
+			}
+			if len(updates[i].Annotations) != 0 {
```

Comment:
This doesn’t allow deleting all annotations; checking for `….Annotations != nil` would make that possible.